Library/Homebrew/os/mac/cache_store.rb

```diff
@@ -0,0 +1,137 @@
+require "dbm"
+require "json"
+
+#
+# `DatabaseCache` acts as an interface to a persistent storage mechanism
+# residing in the `HOMEBREW_CACHE`
+#
+class DatabaseCache
+  # Users have read and write, but not execute permissions
+  DATABASE_MODE = 0666
+
+  # Opens and yields a database in read/write mode
+  #
+  # DBM::WRCREAT: Creates the database if it does not already exist
```

Comment:
I don't think this is the place to document `DBM::WRCREAT`; it'll go in the public API docs but is an implementation detail. Either comment inside the function or consider `create_if_not_exist_mode = DBM::WRCREAT` or similar.